engine/crates/engine-v2/src/engine/execute/single.rs

```diff
@@ -59,29 +62,56 @@ impl<R: Runtime> Engine<R> {
 
             // After recording all operation metadata
             tracing::debug!("Executed operation");
-            response
+
+            (response, on_operation_response_output)
         }
         .instrument(span.clone())
         .await
     }
 }
 
 impl<'ctx, R: Runtime> PreExecutionContext<'ctx, R> {
-    async fn execute_single(mut self, request: Request) -> (Option<OperationMetricsAttributes>, Response) {
-        let operation_plan = match self.prepare_operation(request).await {
+    async fn execute_single(
+        mut self,
+        request: Request,
+    ) -> (Option<OperationMetricsAttributes>, Option<Vec<u8>>, Response) {
```

Comment:
I would strongly consider here and before returning a struct. I would be fine with a tuple if we have a newtype, but a `Vec<u8>` without a field name is very ambiguous